src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseEnvironmentMembers.cs

```diff
@@ -0,0 +1,134 @@
+                    !compilationContext.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemDiagnosticsProcessModule, out var processModuleType) ||
+                    !compilationContext.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemThreadingThread, out var threadType))
+                {
+                    return;
+                }
+
+                // Get the various members needed from Environment, Process, ProcessModule, and Thread.
+                var environmentProcessIdSymbol = environmentType.GetMembers("ProcessId").FirstOrDefault();
+                var environmentProcessPathSymbol = environmentType.GetMembers("ProcessPath").FirstOrDefault();
+                var environmentCurrentManagedThreadIdSymbol = environmentType.GetMembers("CurrentManagedThreadId").FirstOrDefault();
+                var processGetCurrentProcessSymbol = processType.GetMembers("GetCurrentProcess").FirstOrDefault();
+                var processIdSymbol = processType.GetMembers("Id").FirstOrDefault();
+                var processMainModuleSymbol = processType.GetMembers("MainModule").FirstOrDefault();
+                var processModuleFileNameSymbol = processModuleType.GetMembers("FileName").FirstOrDefault();
+                var threadCurrentThreadSymbol = threadType.GetMembers("CurrentThread").FirstOrDefault();
+                var threadManagedThreadId = threadType.GetMembers("ManagedThreadId").FirstOrDefault();
+
+                // This one analyzer looks for multiple things, so figure out which we have the symbols to enable.
+                bool lookForProcessId = processGetCurrentProcessSymbol != null && processIdSymbol != null && environmentProcessIdSymbol != null;
+                bool lookForProcessPath = processGetCurrentProcessSymbol != null && processMainModuleSymbol != null && processModuleFileNameSymbol != null && environmentProcessPathSymbol != null;
+                bool lookForCurrentManagedThreadId = threadCurrentThreadSymbol != null && threadManagedThreadId != null && environmentCurrentManagedThreadIdSymbol != null;
+                if (!lookForProcessId && !lookForProcessPath && !lookForCurrentManagedThreadId)
+                {
+                    return;
+                }
+
+                // Everything we're looking for is a property, so find all property references.
+                compilationContext.RegisterOperationAction(operationContext =>
+                {
+                    var initialPropRef = (IPropertyReferenceOperation)operationContext.Operation;
+
+                    // Process.GetCurrentProcess().Id
+                    if (lookForProcessId && processIdSymbol!.Equals(initialPropRef.Property))
+                    {
+                        if (initialPropRef.Instance is IInvocationOperation getCurrentProcess &&
+                            processGetCurrentProcessSymbol!.Equals(getCurrentProcess.TargetMethod))
```

Comment:
nit: `SymbolEqualityComparer.Default.Equals(symbol1, symbol2)` will avoid the suppression.